### DISCUSSION TITLE
Shareddata not being mounted on reboot and local browser not starting after update

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#734] [Fixed Issues] [answered]
RPi 3B+ with 7in Official Touch Screen, base i20211203 updated to 20220107

It seems this update is having a problem when the local browser is in use.

journalctl -xe reports PHP errors starting at line 824 (n attached journalctl.txt file) and repeated many times thereafter with differing error messages

Jan 07 13:21:15 RAConservatory nginx[426]: 2022/01/07 13:21:15 [error] 426#426: *6 FastCGI sent in stderr: "PHP message: PHP Warning:  scandir(/srv/http/data/bookmarks): Failed to open directory: No such file or directory in /srv/http/index-body.php on line 78PHP message: PHP Warning:  scandir(): (errno 2): No such file or directory in /srv/http/index-body.php on line 78PHP message: PHP Fatal error:  Uncaught TypeError: array_slice(): Argument #1 ($array) must be of type array, bool given in /srv/http/index-body.php:78

These errors seem to be stopping the new 'shareddata' area being mounted automatically on reboot although it appears to be correct in the /etc/fstab file. if I manually mount the shareddata mount point and restart the localbrowser service then system becomes useable again.

Chri
[journalctl.txt](https://github.com/rern/rAudio-1/files/7829921/journalctl.txt)
s

**Comment by ceejayemm** [comment]
Issue title changed to better identify problems

It seems that following this update all my systems where the update had been applied and the Shared Data option enabled work Ok UNTIL a reboot is needed. After the reboot the shareddata mount point is not available and so the system fails to start properly. As noted above the /etc/fstab file seems to be configured properly

PARTUUID=ca59f135-01  /boot  vfat  defaults  0  0
PARTUUID=ca59f135-02  /      ext4  defaults  0  0
192.168.86.250:/export/music/iTunes/Music  /mnt/MPD/NAS/NAS_Music  nfs  defaults,noauto,bg,soft,timeo=5,rw  0  0
192.168.86.250:/export/music/database/RAdb  /srv/http/shareddata  nfs  defaults,noauto,bg,soft,timeo=5  0  0

but the new mount point is not created (even if I run mount -a):

RAOffice:~  #  df -h
Filesystem                                 Size  Used Avail Use% Mounted on
dev                                        446M     0  446M   0% /dev
run                                        482M  488K  481M   1% /run
/dev/mmcblk0p2                              29G  2.8G   25G  11% /
tmpfs                                      482M   12K  482M   1% /dev/shm
tmpfs                                      4.0M     0  4.0M   0% /sys/fs/cgroup
tmpfs                                      482M     0  482M   0% /tmp
/dev/mmcblk0p1                              99M   37M   62M  38% /boot
192.168.86.250:/export/music/iTunes/Music  3.6T  721G  2.9T  20% /mnt/MPD/NAS/NAS_Music

However if I manually mount the share folder using:

mount 192.168.86.250:/export/music/database/RAdb  /srv/http/shareddata

then mount point is created:

RAOffice:/etc  #  df -h
Filesystem                                  Size  Used Avail Use% Mounted on
dev                                         446M     0  446M   0% /dev
run                                         482M  488K  481M   1% /run
/dev/mmcblk0p2                               29G  2.8G   25G  11% /
tmpfs                                       482M   12K  482M   1% /dev/shm
tmpfs                                       4.0M     0  4.0M   0% /sys/fs/cgroup
tmpfs                                       482M     0  482M   0% /tmp
/dev/mmcblk0p1                               99M   37M   62M  38% /boot
192.168.86.250:/export/music/iTunes/Music   3.6T  721G  2.9T  20% /mnt/MPD/NAS/NAS_Music
192.168.86.250:/export/music/database/RAdb  3.6T  721G  2.9T  20% /srv/http/shareddata

and the system recovers (but now seems to be doing a full library update).

Chris

**Comment by ceejayemm** [comment]
@rern 

I think part of this problem might be that the NFS share setup dialogue box entries seem to have changed behind the scenes. I dropped the Music library share  (which has been setup via NFS for as long as I can remember and has never given me any problems) and tried to recreate it via the System / Add Network Storage NFS dialogue. The name of this dialogue has changed to Shared Data (?) and it failed with the error 'IP NAS_Music Not Found'.  It seems to be substituting the Name for the IP number I entered. Are the Add Network Storage and Shared Data dialogue boxes shared and getting mixed up in the process ?

This doesn't solve the mystery of why, with a correct /etc/fstab, the NFS mounts are not being completed correctly on reboot but might give clue if something isn't right behind the scenes.

All of the above seems to be caused by the 20220107 update as if I revert to the (vanilla) i20211203 image everything works as I would expect.

Chris

**Reply by rern** [reply]
Hi Chris,

Please try the UPDATE branch.

**Comment by ceejayemm** [comment, marked as answer]
@rern 

Thanks, the Update branch seems better.  The Network Storage dialogue now seems separate from the Shared Data dialogue - it would be useful to be able to update the Shared Data settings , say if an error has been made.

I am not sure the shared library update is working correctly though.  On my Office system I have 18/20 web radio stations. I deleted the original shared data files from yesterday and allowed the Office system to recreate the shared data.  I then update another of my systems (Test1) to use the revise Update branch version. This system had 6 radio stations setup. After attaching to the share data (no, I did not tell this machine to share its data) the original Office system now had 6 radio stations (?) as did the Test1 system.  If I add a new radio station to the Office system it is not replicated to the Test 1 system which brings me back to my original questions in #731 How do I update the Library on systems which are not the main source of the shared data.

Thanks

Chris

**Comment by ceejayemm** [comment]
@rern 

Spoke too soon.  I have just applied the Update branch to my RPi which has the TouchScreen installed. Update went OK, no other changes made. I then rebooted but the reboot failed.  I cannot get into this system even via SSH.  The display repeatedly shows the RAudio logo but never gets to finish the boot process.

I am reflashing the USB card to i20211203 and will then reapply the 20220107 update branch to that to see what happens.

Chris

**Comment by ceejayemm** [comment]
@rern 

Re-flashed USB drive to i20211202 (I boot this particular RPi from a USB flash drive) and reapplied Update branch of 20220107.  Flashing TouchScreen was cured by adding:

dtparam=i2c_vc_baudrate=50000

to /boot/config.txt

Reconfigured system from scratch (not from backup file) including enabling System / Shared Data option.  By default when attaching Network Storage (NFS) the option to Update Library on Mount is enabled.  The causes the existing Shared Data to be overwritten by this update (despite not enabling the 'Use Data from this system' option on System / Shared Data). This explains why the Network Radios are getting replaced , as noted above.

So I think the update issues for this ticket are now resolved by the latest Update branch version.

I will continue the issues with the Shared Data updates on #731 as this is probably a better place to discuss this rather than continue here.

Thanks

Chris
